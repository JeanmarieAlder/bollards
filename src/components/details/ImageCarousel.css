.carousel-container {
    position: relative;
  }
  
  .carousel-image-div {
    background-color: #e1ede3;
    border-radius: 15px;
    overflow: hidden;
    max-width: 100%;
    max-height: calc(100vh - 60px);
  }
  
  .carousel-image {
    max-width: 100%;
    max-height: calc(100vh - 60px);
  }
  
  .carousel .slide img {
    width: auto;
    vertical-align: top;
    border: 0;
  }
  
  .thumbs-wrapper {
    text-align: center;
  }
  
  .pause-symbol {
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    font-size: 2em;
    color: #000;
    background-color: rgba(255, 255, 255, 0.5); /* White half transparent background */
    padding: 10px; /* Padding around the text */
    border-radius: 10px; /* Border radius */
    opacity: 1; /* Initial opacity */
    transition: opacity 4s ease-in-out; /* Fade out over 2 seconds */
  }
  
  /* Optional: Add a keyframe animation for a smoother fade effect */
  @keyframes fadeOut {
    from {
      opacity: 1;
    }
    to {
      opacity: 0;
    }
  }
  
  .pause-symbol.fade-out {
    animation: fadeOut 4s ease-in-out;
  }
  