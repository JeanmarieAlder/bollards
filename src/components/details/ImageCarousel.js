import React from 'react';
import { Carousel } from 'react-responsive-carousel';
import Config from '../../Config';
import "react-responsive-carousel/lib/styles/carousel.min.css"; // requires a loader
import './ImageCarousel.css';
import { useState, useEffect } from 'react';
import { useTranslation } from "react-i18next";


const ImageCarousel = ({images}) => {
    const { t } = useTranslation();
    const [autoPlay, setAutoPlay] = useState(true);
    const [showPauseSymbol, setShowPauseSymbol] = useState(false);

    const handleInteraction = () => {
        if (autoPlay) {
          // User has interacted, disable autoPlay
          setAutoPlay(false);
          setShowPauseSymbol(true);
        }
      };

      useEffect(() => {
        // After 2 seconds, hide the pause symbol
        const timeoutId = setTimeout(() => {
          setShowPauseSymbol(false);
        }, 3000);
    
        // Cleanup the timeout on component unmount or when autoPlay is re-enabled
        return () => clearTimeout(timeoutId);
      }, [showPauseSymbol, autoPlay]);

    return (
        <div className="carousel-container" onClick={handleInteraction}>
            <Carousel 
                swipeable={true}
                emulateTouch={true} 
                autoPlay={autoPlay} 
                infiniteLoop={true}
                interval={4000} 
                stopOnHover={false}
                preventMovementUntilSwipeScrollTolerance={true}
                swipeScrollTolerance={35}
                onSwipeEnd={handleInteraction}
                >
                    
                {
                    images.map((img, i) => {
                        return (
                            <div key={i} className={'carousel-image-div'}>
                                <img 
                                    className={'carousel-image'} 
                                    src={`${Config.STATIC_FILES_URL}/static/img/bollards/${img}`} 
                                    alt='bollard' />
                            </div>
                        )
                    })
                }
            </Carousel>
            {showPauseSymbol && (
                <div className="pause-symbol fade-out">{t("s_stop_auto_scroll")}</div>
            )}
        </div>
        
    )
}

export default ImageCarousel;