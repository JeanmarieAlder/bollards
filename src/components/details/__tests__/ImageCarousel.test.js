import React from 'react';
import ImageCarousel from '../ImageCarousel';
import renderer from 'react-test-renderer';


test("Component displays correctly", () => {
    const component = renderer.create(
        <ImageCarousel images={['b9909a2f9d33190c.jpg']}/>
      );
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
});

test("Component displays correctly with multiple pictures", () => {
    const component = renderer.create(
        <ImageCarousel images={['b9909a2f9d33190c.jpg', '4e5809202f8beb95.jpg']}/>
      );
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
});