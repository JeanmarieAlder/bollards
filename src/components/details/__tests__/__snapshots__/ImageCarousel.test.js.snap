// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component displays correctly 1`] = `
<div
  className="carousel-container"
  onClick={[Function]}
>
  <div
    className="carousel-root"
  >
    <div
      className="carousel carousel-slider"
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <ul
        className="control-dots"
      >
        <li
          aria-label="slide item 1"
          className="dot selected"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
          value={0}
        />
      </ul>
      <button
        aria-label="previous slide / item"
        className="control-arrow control-prev control-disabled"
        onClick={[Function]}
        type="button"
      />
      <div
        className="slider-wrapper axis-horizontal"
        style={Object {}}
      >
        <ul
          className="slider animated"
          style={
            Object {
              "MozTransitionDuration": "350ms",
              "OTransform": "translate3d(-100%,0,0)",
              "OTransitionDuration": "350ms",
              "WebkitTransform": "translate3d(-100%,0,0)",
              "WebkitTransitionDuration": "350ms",
              "msTransform": "translate3d(-100%,0,0)",
              "msTransitionDuration": "350ms",
              "transform": "translate3d(-100%,0,0)",
              "transitionDuration": "350ms",
            }
          }
        >
          <li
            className="slide selected previous"
            onClick={[Function]}
            style={Object {}}
          >
            <div
              className="carousel-image-div"
            >
              <img
                alt="bollard"
                className="carousel-image"
                src="/static/img/bollards/b9909a2f9d33190c.jpg"
              />
            </div>
          </li>
          <li
            className="slide selected previous"
            onClick={[Function]}
            style={Object {}}
          >
            <div
              className="carousel-image-div"
            >
              <img
                alt="bollard"
                className="carousel-image"
                src="/static/img/bollards/b9909a2f9d33190c.jpg"
              />
            </div>
          </li>
        </ul>
      </div>
      <button
        aria-label="next slide / item"
        className="control-arrow control-next control-disabled"
        onClick={[Function]}
        type="button"
      />
      <p
        className="carousel-status"
      >
        1 of 1
      </p>
    </div>
    <div
      className="carousel"
    >
      <div
        className="thumbs-wrapper axis-vertical"
      >
        <button
          aria-label="previous slide / item"
          className="control-arrow control-prev control-disabled"
          onClick={[Function]}
          type="button"
        />
        <ul
          className="thumbs animated"
          style={
            Object {
              "MozTransform": "translate3d(0,0,0)",
              "MozTransitionDuration": "350ms",
              "MsTransform": "translate3d(0,0,0)",
              "MsTransitionDuration": "350ms",
              "OTransform": "translate3d(0,0,0)",
              "OTransitionDuration": "350ms",
              "WebkitTransform": "translate3d(0,0,0)",
              "WebkitTransitionDuration": "350ms",
              "msTransform": "translate3d(0,0,0)",
              "msTransitionDuration": "350ms",
              "transform": "translate3d(0,0,0)",
              "transitionDuration": "350ms",
            }
          }
        >
          <li
            aria-label="slide item 1"
            className="thumb selected"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            style={
              Object {
                "width": 80,
              }
            }
            tabIndex={0}
          >
            <img
              alt="bollard"
              className="carousel-image"
              src="/static/img/bollards/b9909a2f9d33190c.jpg"
            />
          </li>
        </ul>
        <button
          aria-label="next slide / item"
          className="control-arrow control-next control-disabled"
          onClick={[Function]}
          type="button"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Component displays correctly with multiple pictures 1`] = `
<div
  className="carousel-container"
  onClick={[Function]}
>
  <div
    className="carousel-root"
  >
    <div
      className="carousel carousel-slider"
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <ul
        className="control-dots"
      >
        <li
          aria-label="slide item 1"
          className="dot selected"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
          value={0}
        />
        <li
          aria-label="slide item 2"
          className="dot"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
          value={1}
        />
      </ul>
      <button
        aria-label="previous slide / item"
        className="control-arrow control-prev"
        onClick={[Function]}
        type="button"
      />
      <div
        className="slider-wrapper axis-horizontal"
        style={Object {}}
      >
        <ul
          className="slider animated"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "MozTransitionDuration": "350ms",
              "OTransform": "translate3d(-100%,0,0)",
              "OTransitionDuration": "350ms",
              "WebkitTransform": "translate3d(-100%,0,0)",
              "WebkitTransitionDuration": "350ms",
              "msTransform": "translate3d(-100%,0,0)",
              "msTransitionDuration": "350ms",
              "transform": "translate3d(-100%,0,0)",
              "transitionDuration": "350ms",
            }
          }
        >
          <li
            className="slide"
            onClick={[Function]}
            style={Object {}}
          >
            <div
              className="carousel-image-div"
            >
              <img
                alt="bollard"
                className="carousel-image"
                src="/static/img/bollards/4e5809202f8beb95.jpg"
              />
            </div>
          </li>
          <li
            className="slide selected previous"
            onClick={[Function]}
            style={Object {}}
          >
            <div
              className="carousel-image-div"
            >
              <img
                alt="bollard"
                className="carousel-image"
                src="/static/img/bollards/b9909a2f9d33190c.jpg"
              />
            </div>
          </li>
          <li
            className="slide"
            onClick={[Function]}
            style={Object {}}
          >
            <div
              className="carousel-image-div"
            >
              <img
                alt="bollard"
                className="carousel-image"
                src="/static/img/bollards/4e5809202f8beb95.jpg"
              />
            </div>
          </li>
          <li
            className="slide selected previous"
            onClick={[Function]}
            style={Object {}}
          >
            <div
              className="carousel-image-div"
            >
              <img
                alt="bollard"
                className="carousel-image"
                src="/static/img/bollards/b9909a2f9d33190c.jpg"
              />
            </div>
          </li>
        </ul>
      </div>
      <button
        aria-label="next slide / item"
        className="control-arrow control-next"
        onClick={[Function]}
        type="button"
      />
      <p
        className="carousel-status"
      >
        1 of 2
      </p>
    </div>
    <div
      className="carousel"
    >
      <div
        className="thumbs-wrapper axis-vertical"
      >
        <button
          aria-label="previous slide / item"
          className="control-arrow control-prev control-disabled"
          onClick={[Function]}
          type="button"
        />
        <ul
          className="thumbs animated"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "MozTransform": "translate3d(0,0,0)",
              "MozTransitionDuration": "350ms",
              "MsTransform": "translate3d(0,0,0)",
              "MsTransitionDuration": "350ms",
              "OTransform": "translate3d(0,0,0)",
              "OTransitionDuration": "350ms",
              "WebkitTransform": "translate3d(0,0,0)",
              "WebkitTransitionDuration": "350ms",
              "msTransform": "translate3d(0,0,0)",
              "msTransitionDuration": "350ms",
              "transform": "translate3d(0,0,0)",
              "transitionDuration": "350ms",
            }
          }
        >
          <li
            aria-label="slide item 1"
            className="thumb selected"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            style={
              Object {
                "width": 80,
              }
            }
            tabIndex={0}
          >
            <img
              alt="bollard"
              className="carousel-image"
              src="/static/img/bollards/b9909a2f9d33190c.jpg"
            />
          </li>
          <li
            aria-label="slide item 2"
            className="thumb"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            style={
              Object {
                "width": 80,
              }
            }
            tabIndex={0}
          >
            <img
              alt="bollard"
              className="carousel-image"
              src="/static/img/bollards/4e5809202f8beb95.jpg"
            />
          </li>
        </ul>
        <button
          aria-label="next slide / item"
          className="control-arrow control-next control-disabled"
          onClick={[Function]}
          type="button"
        />
      </div>
    </div>
  </div>
</div>
`;
